src/Dapr.Client/DaprClient.cs

```diff
@@ -14,11 +14,21 @@ namespace Dapr.Client
     using Google.Protobuf;
 
     /// <summary>
+    /// <para>
     /// Defines client methods for interacting with Dapr endpoints.
-    /// Use <see cref="DaprClientBuilder"/> to create <see cref="DaprClient"/>
+    /// Use <see cref="DaprClientBuilder"/> to create <see cref="DaprClient"/>.
+    /// </para>
+    /// <para>
+    /// Implementations of <see cref="DaprClient" /> implement <see cref="IDisposable" /> because the client
+    /// accesses network resources. For best performance, create a single long-lived client instance and share
+    /// it for the lifetime of the application. Avoid creating and disposing a client instance for each operation
+    /// that the application performs - this can lead to socket exhaustion and other problems.
+    /// </para>
     /// </summary>
-    public abstract class DaprClient
+    public abstract class DaprClient : IDisposable
```

Comment:
Isn't a similar change needed in Actors code?

Reply:
No, I don't think so - actors uses a static instance of HttpClient. It's not ideal, but it's not broken either right now.